Exercise_1.java

```diff
@@ -1,21 +1,33 @@
-class BinarySearch { 
-    // Returns index of x if it is present in arr[l.. r], else return -1 
-    int binarySearch(int arr[], int l, int r, int x) 
-    { 
-        //Write your code here
-    } 
-  
-    // Driver method to test above 
-    public static void main(String args[]) 
-    { 
-        BinarySearch ob = new BinarySearch(); 
-        int arr[] = { 2, 3, 4, 10, 40 }; 
-        int n = arr.length; 
-        int x = 10; 
-        int result = ob.binarySearch(arr, 0, n - 1, x); 
-        if (result == -1) 
-            System.out.println("Element not present"); 
+class BinarySearch {
+    // Returns index of x if it is present in arr[l.. r], else return -1
+    int binarySearch(int arr[], int l, int r, int x) {
+        // Write your code here
+        int mid = (l + r) / 2; // Finding the mid index of the array
+        // check to see if left index is less than right one. if so then element is not
+        // present
+        if (l < r) {
+            if (x == arr[mid]) // checking if the element is at middle. If so then returning index value

```

Comment:
Is this solution working for some test cases that you wrote. it should be` l <= r` if you are using `low = mid + 1 `and `high = mid -1.` Test again please

Reply:
okay. corrected the code